doc/ref_impl/cntt-ri/chapters/chapter05.md

```diff
@@ -5,15 +5,280 @@
 
 ## Table of Contents
 * [5.1 Introduction](#5.1)
-* [5.2 Heading](#5.2)
-* [5.3 Heading](#5.3)
+* [5.2 Installer requirements](#5.2)
+   * [5.2.1 General](#5.2.1)
+   * [5.2.2 Additional](#5.2.2)
+* [5.3 Descriptor file definition](#5.3)
+   * [5.3.1 Hardware resource information](#5.3.1)
+   * [5.3.2 Server template](#5.3.2)
+   * [5.3.3 Server information](#5.3.3)
+   * [5.3.4 Software configuration definition](#5.3.4)
+   * [5.3.5 Network informations](#5.3.5)
+   * [5.3.6 Controller informations](#5.3.6)
+   * [5.3.7 Compute informations](#5.3.7)
+   * [5.3.8 Distributed storage informations](#5.3.8)
+   * [5.3.9 NTP server informations](#5.3.9)
+   * [5.3.10 DNS server informations](#5.3.10)
+   * [5.3.11 Deployment host informations](#5.3.11)
+   * [5.3.12 Deployment control informations](#5.3.12)
+   * [5.3.13 Proxy informations](#5.3.13)   
 
 <a name="5.1"></a>
 ## 5.1 Introduction
 
+**must**: Requirements that are marked as _must_ are considered mandatory and must exist in the reference implementation and implemented by installer. The same applies to _must not_.
+
+**should**: Requirements that are marked as _should_ are expected to be fulfilled by the reference implementation but it is up to each installer to accept an implementation targeting this reference implementation that is not reflecting on any of those requirements. The same applies to _should not_.
+
+**may**: Requirements that are marked as _may_ are considered optional. The same applies to _may not_.
```

Comment:
I see what we're trying to do here with should and may, but it's very confusing to the reader. If I understand the intent, both "must" and "should" are mandatory, the difference being that must is compulsory for everyone and should is only compulsory for the Reference Implementation. 

My suggestion to try to eliminate errors caused by this ambiguity would be to identify them both as Must, and then qualify them to indicate the scope (e.g., all or RI-only). Using sample text from this chapter, use might read:
* "Installer **must-(all)** accept a descriptor file...", and 
* "Descriptor file **must-(RI only)** include hardware resource configuration...". 

It's kind of clunky, but it's clear and unambiguous.

Reply:
Hi Mark, I don't understand the difference between must-all and must-RI. I think all of those description belong to RI only.  So I propose we only keep must and may to explain mandatory and optional.
In this case, the descriptor file must include hardware resource configuraiton and software....
Do you think it's ok?

Reply:
Hi, You're saying eliminate should, so we'll only have must and may. I'm good with that. Let me know when you've had a chance to make the changes, and I'll click "resolve" for this sub-thread. Thanks, -Mark

Reply:
Exactly, check the new commit please.

Reply:
That's MUCH better. Thank you. 

FYI, there are several instances of the word "should" in the chapter. Not the should we just dealt with, but should in the more conversational sense. This is not a request for a change, so taking action is at your discretion, or as the document would say, you _may_ take action on this: words such as "should", "would" and "could" are sub-optimal when used in specifications because they imply their subject is discretionary. Consider using "shall", instead. It makes for a stronger statement.